plugins/ingest/src/main/java/org/elasticsearch/ingest/TemplateService.java

```diff
@@ -0,0 +1,70 @@
+package org.elasticsearch.ingest;
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Map;
+
+public interface TemplateService {
+
+    default Template compile(String template) {
+        if (template != null) {
```

Comment:
not sure about this default impl. Seems like the runtime has to plug in its own template service and template impls. but in the interface we assume that we always have mustache. I don't mean to make the engine pluggable, but I think we have to better figure out who needs to do what and how when the ingest library is used outside of the context of es core. I think for now this assumption would be better placed as part of the implementation.